Structure_Engine/Modify/Flip.cs

```diff
@@ -86,6 +120,87 @@ public static Stem Flip(this Stem stem)
+            {
+                ISectionProperty tempSection = null;
+                IProfile profile = geometricalSection.SectionProfile;
+                IProfile tempProfile = IFlipProfile(profile);
+                IMaterialFragment material = section.Material;
+
+                tempProfile.Name = profile.Name;
+                tempProfile.Fragments = profile.Fragments;
+                tempSection = Create.GenericSectionFromProfile(tempProfile, material, section.Name);
+                return tempSection;
+            }
+            else
+            {
+                Base.Compute.RecordError("The given shape profile is not an IGeometricalSection.");
+                return null;
+            }
+        }
+
+        private static IProfile IFlipProfile(IProfile profile)
+        {
+            if (profile is BoxProfile || profile is ICellularOpening || profile is CircleProfile || profile is FabricatedBoxProfile ||
+                    profile is FabricatedISectionProfile || profile is GeneralisedFabricatedBoxProfile || profile is ISectionProfile || profile is KiteProfile
+                    || profile is RectangleProfile || profile is TaperFlangeISectionProfile || profile is TSectionProfile ||
+                    profile is TubeProfile || profile is VoidedISectionProfile)
+            {
+                return profile;
+            }
+            else
+            {
+                return FlipProfile(profile as dynamic);
+            }
+        }
+        private static IProfile FlipProfile(AngleProfile oldProfile)
+        {
+            return Spatial.Create.AngleProfile(oldProfile.Height, oldProfile.Width, oldProfile.WebThickness, oldProfile.FlangeThickness, oldProfile.RootRadius,
+                oldProfile.ToeRadius, !oldProfile.MirrorAboutLocalZ, oldProfile.MirrorAboutLocalY);
+        }
+
+        private static IProfile FlipProfile(ChannelProfile oldProfile)
+        {
+            return Spatial.Create.ChannelProfile(oldProfile.Height, oldProfile.FlangeWidth, oldProfile.WebThickness, oldProfile.FlangeThickness,
+                oldProfile.RootRadius, oldProfile.ToeRadius, !oldProfile.MirrorAboutLocalZ);
+        }
+        private static IProfile FlipProfile(FreeFormProfile oldProfile)
+        {
+            List<ICurve> curves = oldProfile.Edges.ToList();
+            return Spatial.Create.FreeFormProfile(curves.Select(x => x.IMirror(Geometry.Create.Plane(new Point(), Vector.XAxis))));
+        }
+        private static IProfile FlipProfile(GeneralisedTSectionProfile oldProfile)
+        {
+            return Spatial.Create.GeneralisedTSectionProfile(oldProfile.Height, oldProfile.WebThickness, oldProfile.RightOutstandWidth, oldProfile.RightOutstandThickness,
+                oldProfile.LeftOutstandWidth, oldProfile.LeftOutstandThickness, oldProfile.MirrorAboutLocalY);
+        }
+        private static IProfile FlipProfile(KiteProfile oldProfile)
+        {
+            return oldProfile;
```

Comment:
If this is problematic, might be good to add a warning about it, or let it fallback to the other method